pkg/controller/controller_test.go

```diff
@@ -32,29 +31,26 @@ var _ = Describe("Controller", func() {
 	var (
 		controller *Controller
 		fakeClient *fake.Clientset
+		pvcSource  *k8stesting.FakePVCControllerSource
+		podSource  *k8stesting.FakeControllerSource
 		stop       chan struct{}
 	)
-	type testT struct {
-		descr       string
-		ns          string
-		name        string // name of test pvc
-		qops        operation
-		annotations map[string]string
-		expectError bool
-	}
-
-	setUpInformer := func(pvc *v1.PersistentVolumeClaim, op operation) {
 
-		stop = make(chan struct{})
-		fakeClient = fake.NewSimpleClientset(pvc)
-		pvcSource := k8stesting.NewFakePVCControllerSource()
-		podSource := k8stesting.NewFakeControllerSource()
+	setUpInformers := func(pvc *v1.PersistentVolumeClaim, pod *v1.Pod, ns string, op operation, stop <-chan struct{}) {
+		// Populate the informer caches with test objects
+		if pvc != nil {
```

Comment:
If pvc or pod are nil should we end the tests (or panic)? I know we don't want to reference nil ptrs but is this check merited?